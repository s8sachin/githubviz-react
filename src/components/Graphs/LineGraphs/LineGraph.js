import React, { Component } from 'react';
import { XYPlot,XAxis,YAxis, VerticalGridLines, HorizontalGridLines,Hint,LineMarkSeries
} from 'react-vis';
import { Grid, Row,Col} from 'react-bootstrap';

const data1 = [
    {x: "Repo", y: "C1"},
    {x: "Repo1", y: "C2"},
    {x: "Repo2", y: "C3"},
    {x: "Repo3", y: "C4"},
    {x: "Repo4", y: "C5"},
    {x: "Repo5", y: "C6"},

  ];


class LineGraph extends Component {
    state = {
        value2: false
      }
  render() {
    const {value2} = this.state;
    return (
      <div>
      <Grid>
        <Row>
      <Col xs={12} md={12}>
      <XYPlot height={300} width={500} stroke="orange" xType="ordinal" yType="ordinal" >
        <VerticalGridLines />
        <HorizontalGridLines />
        <XAxis />
        <YAxis />
        <LineMarkSeries data={data1}
         onValueMouseOver={v => this.setState({ value2: v })}
         onSeriesMouseOut={v => this.setState({ value2: false })}
        />
         {value2 && <Hint value={value2} />}
      </XYPlot>
    </Col>
    </Row>
        </Grid>
        </div>
    );
  }
}

export default LineGraph;
