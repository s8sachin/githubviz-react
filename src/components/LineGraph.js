import React, { Component } from 'react';
import Header from './commons/Header';
import 'react-vis/dist/style.css';
import { XYPlot,VerticalBarSeries,XAxis,YAxis, VerticalGridLines, HorizontalGridLines,Hint,LineMarkSeries
  } from 'react-vis';
  import { Grid, Row,Col} from 'react-bootstrap';
  const data = [
    {x: 0, y: 8},
    {x: 1, y: 5},
    {x: 2, y: 4},
    {x: 3, y: 9},
    {x: 4, y: 1},
    {x: 5, y: 7},
    {x: 6, y: 6},
    {x: 7, y: 3},
    {x: 8, y: 2},
    {x: 9, y: 0}
  ];
  const data1 = [
    {x: "Repo", y: "C1"},
    {x: "Repo1", y: "C2"},
    {x: "Repo2", y: "C3"},
    {x: "Repo3", y: "C4"},
    {x: "Repo4", y: "C5"},
    {x: "Repo5", y: "C6"},

  ];
class LineGraph extends Component {
  state = {
    value: false,
    value2: false
  }
  render () {
    const {value} = this.state;
    const {value2} = this.state;
    return (
      <div>
        <Header />
        <Grid>
          <Row className="show-grid">
            <Col xs={12} md={6}>
              <div>
                <XYPlot height={300} width={500} color="orange" stroke="black"  xType="ordinal">
                  <VerticalGridLines />
                  <HorizontalGridLines />
                  <XAxis />
                  <YAxis />
                  <VerticalBarSeries data={data}
                    onValueMouseOver={v => this.setState({ value: v })}
                    onSeriesMouseOut={v => this.setState({ value: false })} />
                  {value && <Hint value={value} />}
                </XYPlot>
                <h3 className="head-custom">BarGragh:</h3>
                <p className="para-custom">Bar series come in two flavors, HorizontalBarSeries and VerticalBarSeries</p>
              </div>
            </Col>
            <Col xs={12} md={6}>
              <XYPlot height={300} width={300} stroke="orange" xType="ordinal" yType="ordinal" >
                <VerticalGridLines />
                <HorizontalGridLines />
                <XAxis />
                <YAxis />
                <LineMarkSeries data={data1}
                 onValueMouseOver={v => this.setState({ value2: v })}
                 onSeriesMouseOut={v => this.setState({ value2: false })}
                />
                 {value2 && <Hint value={value2} />}
              </XYPlot>
              <h3 className="head-custom">LineGragh:</h3>
                <p className="para-custom">Line series come in two flavors, HorizontalBarSeries and VerticalBarSeries</p>
            </Col>
          </Row>
        </Grid>
      </div>
    )
  }
}
export default LineGraph;