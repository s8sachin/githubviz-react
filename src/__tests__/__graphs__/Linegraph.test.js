import React from 'react';
import { shallow,  configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import LineGraph from '../../components/Graphs/LineGraphs/LineGraph';

configure({ adapter: new Adapter() });

describe('LineGraph', () => {
it('should mount Header component', () => {
    const wrapper = shallow(<LineGraph />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})