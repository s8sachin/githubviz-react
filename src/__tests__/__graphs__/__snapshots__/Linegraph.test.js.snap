// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineGraph should mount Header component 1`] = `
<div>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={false}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={12}
        xs={12}
      >
        <XYPlot
          className=""
          height={300}
          stroke="orange"
          width={500}
          xType="ordinal"
          yType="ordinal"
        >
          <VerticalGridLines
            attr="x"
            direction="vertical"
          />
          <HorizontalGridLines
            attr="y"
            direction="horizontal"
          />
          <XAxis
            attr="x"
            attrAxis="y"
            orientation="bottom"
          />
          <YAxis
            attr="y"
            attrAxis="x"
            orientation="left"
          />
          <LineMarkSeries
            className=""
            curve={null}
            data={
              Array [
                Object {
                  "x": "Repo",
                  "y": "C1",
                },
                Object {
                  "x": "Repo1",
                  "y": "C2",
                },
                Object {
                  "x": "Repo2",
                  "y": "C3",
                },
                Object {
                  "x": "Repo3",
                  "y": "C4",
                },
                Object {
                  "x": "Repo4",
                  "y": "C5",
                },
                Object {
                  "x": "Repo5",
                  "y": "C6",
                },
              ]
            }
            getNull={[Function]}
            lineStyle={Object {}}
            markStyle={Object {}}
            onSeriesMouseOut={[Function]}
            onValueMouseOver={[Function]}
            opacity={1}
            stack={false}
            strokeStyle="solid"
            style={Object {}}
          />
        </XYPlot>
      </Col>
    </Row>
  </Grid>
</div>
`;
